core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -374,49 +375,63 @@ private[spark] object PythonRDD extends Logging {
     // The right way to implement this would be to use TypeTags to get the full
     // type of T.  Since I don't want to introduce breaking changes throughout the
     // entire Spark API, I have to use this hacky approach:
```

Comment:
I believe that these three comment lines originally referred to the code on line 381 (now line 395), where we look at the runtime type of the first element of the iterator.  For clarity's sake, I think we should move the new `write` methods above this comment so that it stays in the right spot.